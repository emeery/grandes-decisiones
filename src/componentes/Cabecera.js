import React from 'react';
import {NavLink} from 'react-router-dom';

// <NavLink to="/ayuda" activeClassName='is-active'>Ayuda</NavLink>
const Cabecera = () => (
    <header>
    <h1>Libros de Notas.</h1>
    <h3>La pieza faltante.</h3>
    <NavLink to="/" activeClassName='is-active' exact={true} >Escritorio</NavLink>
    <NavLink to="/crealo" activeClassName='is-active'>Crear Gastos</NavLink>
    
    </header>

); export default Cabecera;

