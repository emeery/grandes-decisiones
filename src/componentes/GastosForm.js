import React from 'react';
import moment from 'moment';
import { SingleDatePicker} from 'react-dates'; 

// const ahora = moment();
// console.log(ahora.format('MMM Do, YYYY') );

class GastosForm extends React.Component {
    constructor(props) {
        super( props );
        this.state = {
        descripcion: props.gasto ?  props.gasto.descripcion : '' , 
        nota: props.gasto ? props.gasto.nota : '' , 
        monto: props.gasto ? (props.gasto.monto / 100 ).toString() : '', 
        date: props.gasto ? moment(props.gasto.date) : moment(),
        calendarFocused : false,
        error: ''
        };
    }
    descripcionCambio = (e) => {
        const descripcion = e.target.value; 
        this.setState(() => ({ descripcion })); 
    };
    // persistncia ();
    notaCambio = (e) => {
        const nota = e.target.value;
        this.setState(()=> ({ nota }));
    };
    montoCambio = (e) => {
        const monto = e.target.value;
        if (!monto || monto.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ monto }));
        }
    };
    onDatesChange = (date) => {
        if(date) {
            this.setState(()=> ({ date }));
        }
    };
    // focused - requerido
    onFocusChange = ({focused}) => {
        this.setState(() => ({ calendarFocused: focused }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.descripcion || !this.state.monto) {
            this.setState(() => ({error: 'ingresa los 2 valores'}));
        } else {
            this.setState(() => ({error: ''}));
            this.props.onSubmit({
                descripcion: this.state.descripcion,
                monto: parseFloat(this.state.monto, 10) * 100,
                date: this.state.date.valueOf(), // nmero regular que representa esa fecha
                nota: this.state.nota
            });
        }
    };
    render() {
        return( <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>      
            <input 
            type='text'
            placeholder='Descripcion'
            autoFocus
            value={this.state.descripcion}
            onChange={this.descripcionCambio} 
            />
            <input
            type='text'
            placeholder='Monto'
            value={this.state.monto}
            onChange={this.montoCambio}
            />
            
            <SingleDatePicker
            date={this.state.date}
            onDateChange={this.onDatesChange}
            focused={this.state.calendarFocused} 
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={()=>false}
            />
            
            <textarea
            placeholder='agrega una nota para tu gasto'
            value={this.state.nota}
            onChange={this.notaCambio}
            >
            </textarea>

            <button>Agrega Gasto</button>
        </form>    
        </div>);
    }
}; export default GastosForm;


// crear playlist.
// nombrePLaylist /* -string, nombre 1/100*/
// <button /*  elegir imagen */
// descripcion /* -string estos tres 0/300 */
// <button /* cancelar */
// <button /* crear */
