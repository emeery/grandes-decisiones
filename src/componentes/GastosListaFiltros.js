import React from 'react';
import {connect} from 'react-redux';
import {DateRangePicker} from 'react-dates'; 

import {
    establecerTextoFiltros, 
    ordenarPorMonto, 
    ordenarPorFecha,
    establecerFechaInicio, 
    establecerFechaFin
} from '../acciones/filtros';

class GastosListaFiltros extends React.Component {
    state = {
        calendarFocused: null
    };
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(establecerFechaInicio(startDate) );
        this.props.dispatch(establecerFechaFin(endDate) );
    };
    onFocusChange = (calendarFocused) => {
        this.setState(() => ({ calendarFocused }));
    }
    render() {
        return  ( 
        <div>
        
        <input 
        type='text' 
        value={this.props.filtros.texto}
        onChange={(e) => {
            this.props.dispatch(establecerTextoFiltros(e.target.value));
        }} />
        
        <select 
        value={this.props.filtros.ordenarPor}
        onChange={(e) => {
            if (e.target.value === 'fecha') {
                this.props.dispatch(ordenarPorFecha() );
            } else if (e.target.value === 'monto') {
                this.props.dispatch(ordenarPorMonto() );
            }
        }}
        >
        <option value='fecha'>
            fecha </option>
        <option value='monto'>
            monto </option>
        
        </select>
        <DateRangePicker
        startDate={this.props.filtros.startDate}
        endDate={this.props.filtros.endDate}
        onDatesChange={this.onDatesChange}
        focusedInput={this.state.calendarFocused}
        onFocusChange={this.onFocusChange}
        showClearDates={true}
        numberOfMonths={1}
        isOutsideRange={() => false}
        />
    </div> );
    }
};

const mapStateToProps = (state) => {
    return { filtros: state.filtros };
};

export default connect(mapStateToProps)(GastosListaFiltros);