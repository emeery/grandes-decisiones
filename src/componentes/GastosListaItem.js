import React from 'react';
import { Link } from 'react-router-dom';
//import {connect } from 'react-redux';
// import {removerGasto} from '../acciones/gastos'; connect 

// destructuracion dispatch
        
//dispatch, id, descripcion, monto, fecha
const GastosListaItem = ({
    id, descripcion, monto, date}) => (
    <div>
        <Link to={`/edita/${id}`}>
        <h3> {descripcion} </h3>
        </Link>
        <h2> {monto} - {date}</h2>
       
    </div>
);
export default GastosListaItem;

// const GastosListaItem = ({ 
//      dispatch, id, descripcion, monto, fecha}) => (
//      <div>
//         <h3> descripcion </h3>
//         <h2>fecha: {fecha} - monto: {monto}</h2>
//         <button onClick={() => {
//             dispatch(removeGasto({ id }));
//         }}
//         > remover </button>
//     </div>
// );
// export default connect()(GastosListaItem);

// <h2> monto fecha </h2>
// <button onClick={() => {
//     dispatch(removerGasto({ id }));
// }}
// > remover </button>