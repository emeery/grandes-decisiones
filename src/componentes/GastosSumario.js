import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import seleccionaGastos from '../selectores/gastos'; // creadoEnElMomento
import seleccionaGastosTotal from '../selectores/gastos-totales';

export const GastosSumario = ({ gastosContador, gastosTotal}) => {
    //const sinGastos = gastosContador === 0 ? 'no hay gastos' : gastosContador;
    const gastosPalabra = gastosContador === 1 ? 'Gasto' : 'Gastos';
    //gastosPalabra === 0 ? '' : gastosContador;
    const formatoGastosTotal = numeral(gastosTotal / 100).format('$0,0.00'); 
    
    return(<div>
        <h2> 
        Viendo -
        {gastosContador}
        {gastosPalabra} -
        Total - 
        {formatoGastosTotal}</h2>
        </div>);
};

const mapStatetoProps = (state) => {
    const gastosVisibles = seleccionaGastos(state.gastos, state.filtros );
    return {
        gastosContador: gastosVisibles.length,
        gastosTotal: seleccionaGastosTotal(gastosVisibles)
    };
};

export default connect(mapStatetoProps)(GastosSumario)