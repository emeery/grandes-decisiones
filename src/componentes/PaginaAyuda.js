import React from 'react';

const PaginaAyuda = () => (
    <div><h1>PÃ¡gina Ayuda</h1></div>
); export default PaginaAyuda;