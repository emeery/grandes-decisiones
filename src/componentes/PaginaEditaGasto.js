import React from 'react';
import {connect} from 'react-redux';
import GastosForm from './GastosForm';
import {
    editarGasto, 
    removerGasto 
} from '../acciones/gastos';

const PaginaEditaGasto = (props) => {
    return ( 
        <div>
        <GastosForm
        gasto ={ props.gasto }
        onSubmit ={( gasto ) => {
            props.dispatch( editarGasto(props.gasto.id, gasto));
            props.history.push('/');
        }}
        />
        <button onClick={() => {
            props.dispatch( removerGasto({id: props.gasto.id}));
            props.history.push('/')
        }}> X </button>
    </div> );
};
// subconjunto 
const mapStateToProps = (state, props) => {
    return { 
        gasto: state.gastos.find((gasto) => gasto.id === props.match.params.id) 
    };
};

export default connect(mapStateToProps)(PaginaEditaGasto);

