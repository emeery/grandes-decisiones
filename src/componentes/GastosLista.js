import React from 'react';
import { connect } from 'react-redux';
import GastosListaItem from './GastosListaItem';
import seleccionaGastos from '../selectores/gastos';

// conecta tu componente a la tienda redux
const GastosLista = (props) => (
    <div><h1>Expendio Lista</h1>
        {props.gastos.length}
        {props.gastos.map((gastitos) => {
           return <GastosListaItem 
                key={gastitos.id} {...gastitos}/>
        })}
    </div>
); 
// definimos lo que queremos de la tienda
// y definimos el componnte GastosLista 

const mapStatetoProps = (state) => {
    return { 
        gastos: seleccionaGastos(state.gastos, state.filtros)
    };
};
export default connect(mapStatetoProps)(GastosLista);
