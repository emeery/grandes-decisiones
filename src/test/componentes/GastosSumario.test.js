import React from 'react';
import {shallow } from 'enzyme';
import {GastosSumario} from '../../componentes/GastosSumario';

test('componente GastosSumario con un gasto', ()=>{
    const envoltura = shallow(
        <GastosSumario
            gastosContador={1}
            gastosTotal={120}
        />);
    expect(envoltura).toMatchSnapshot();
});

test('componente GastosSumario con multiples gastos', ()=>{
    const envoltura = shallow(
        <GastosSumario
            gastosContador={2}
            gastosTotal={120932929}
        />);
    expect(envoltura).toMatchSnapshot();
});