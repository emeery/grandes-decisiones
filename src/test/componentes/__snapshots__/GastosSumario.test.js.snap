// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente GastosSumario con multiples gastos 1`] = `
<div>
  <h2>
    Viendo -
    2
    Gastos
     - Total -
    $1,209,329.29
  </h2>
</div>
`;

exports[`componente GastosSumario con un gasto 1`] = `
<div>
  <h2>
    Viendo -
    1
    Gasto
     - Total -
    $1.20
  </h2>
</div>
`;
